### PR TITLE
[terminal] Send proper cols and rows when first connecting to the terminal process

### PR DESCRIPTION
**Pull request by svenefftinge**
The issue here is that the terminal process is always initialized with the defaults of 80 * 24. We need to make sure that the xterm is opened and a resize event came in, before starting/attaching to the backend process, so that we can immediately call resize with proper numbers.

Signed-off-by: Sven Efftinge <sven.efftinge@typefox.io>

### CODE REVIEW

**Comment by svenefftinge** [comment]
Fixes https://github.com/theia-ide/theia/issues/1330

File: packages/terminal/src/browser/terminal-widget.ts [hunk 1]
```diff
@@ -54,17 +55,19 @@ export class TerminalWidget extends BaseWidget implements StatefulWidget {
...
     private endpoint: Endpoint;
     protected restored = false;
     protected closeOnDispose = true;
-    protected waitForStarted: Promise<void>;
-    protected started: Function;
     protected openAfterShow = false;
     protected isOpeningTerm = false;
     protected isTermOpen = false;
 
+    protected waitForStarted = new Deferred<void>();
```
Comment (marcdumais-work) [inline packages/terminal/src/browser/terminal-widget.ts]: I do not see that we wait for this promise anywhere. Should it be added in the ```Promise.all()``` in ```doResize()```?

Comment (svenefftinge) [reply packages/terminal/src/browser/terminal-widget.ts]: No it is no longer neccessary. I will remove it thanks.

**Comment by svenefftinge** [comment]
@marcdumais-work I removed the unused promise. Did you test the changes already?

**Comment by marcdumais-work** [comment]
yes, looks good. Thanks Sven

**Pull request merged**